### PR TITLE
Reflectometry Preview tab Edit ROI

### LINKED ISSUE #34174: Add edit button to the region selector toolbar
Following #34173 it is now possible to add a simple rectangular ROI to the ISIS reflectometry interface Preview tab. This is currently very basic functionality and does not interact well with buttons on the sliceviewer toolbar. To improve this, we need to:

- [x] Add an Edit button that allows the user to modify the existing ROI(s) similarly to the embedded instrument viewer widget.
- [x] Add visual indication of which ROI is currently selected
- [x] Allow deletion of an ROI by selecting it and pressing the Del button
- [x] The grab handles are a bit awkward to use because you need to click inside the rectangle to select it and therefore the portion of the handles that is outside the area don't work. It is a bit awkward to fix but we could maybe increase the size of the selection area, or reduce the grab handle size
- [x] Set the cursor when hovering over a region. This will make it clearer when you are able to resize/move the region

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR makes it possible to edit multiple different rectangle ROI's when the toolbar Edit button is selected. It also connects up the toolbar to give a visual indication for which tool you have selected.

**To test:**
1. Open ISIS Reflectometry interface
2. Load `INTER45455_inst`, and change angle to 1.00
3. Mess around with the toolbar and try edit/draw regions of interests. Make sure the behaviour is the same as seen on the inst view.

Refs #34174

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewModel.cpp [hunk 1]
```diff
@@ -106,14 +106,20 @@ ProcessingInstructions PreviewModel::getProcessingInstructions() const {
 
 void PreviewModel::setSelectedRegion(Selection const &selection) {
   // TODO We will need to allow for more complex selections, but for now the selection just consists two y indices
```
Comment (thomashampson) [inline qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewModel.cpp:108]:
```suggestion
  // TODO We will need to allow for more complex selections, but for now the selection just consists two y indices per rectangle selection
```

File: qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py [hunk 2]
```diff
@@ -64,22 +62,90 @@ def test_update_workspace_updates_view(self):
...
+        selector_two = Mock()
+        selector_two.extents = [x3, x4, y3, y4]
+
+        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector._selectors.append(selector_one)
+        region_selector._selectors.append(selector_two)
+
+        region = region_selector.get_region()
+        self.assertEqual(4, len(region))
+        self.assertEqual([y1, y2, y3, y4], region)
```
Comment (thomashampson) [inline qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py:103]:
```suggestion
        region_selector, selector_one, selector_two = self._mock_selectors()

        region = region_selector.get_region()
        self.assertEqual(4, len(region))
        self.assertEqual([selector_one.extents[1], selector_one.extents[3], selector_two.extents[1], selector_two.extents[3]], region)
```

**Review by thomashampson — COMMENTED** [review]
Two very minor suggestions that can be addressed in the next PR...

**Comment by robertapplin** [comment]
I've made those changes in this PR seeing as the tests were failing anyway. The tests were failing because `drag_from_anywhere` and `ignore_event_outside` RectangleSelector kwargs are only available for matplotlib >= v3.5

**Review by gemmaguest — APPROVED** [review]
Code looks good and functions as expected :+1:. There are some quirks such as the grab handles being a bit fiddly to use but this can be looked at in a future PR seeing as this is part of work in progress that is not available in the release build yet.

**Pull request merged**